modules/database/src/main/java/com/intuit/wasabi/database/impl/DBITransaction.java

```diff
@@ -56,29 +56,31 @@ public DBITransaction(DBI dbi) {
      */
     @Override
     public Object transaction(Block block) {
+        Handle handle = null;
         try {
-            begin();
+            handle = getHandle();
+            begin(handle);
             Object value = block.value(this);
-            commit();
+            commit(handle);
             return value;
         } catch (WasabiException | IllegalArgumentException e) {
             LOGGER.warn("Problem executing block. Trying to rollback...", e);
-            tryRollback();
+            tryRollback(handle);
             throw e;
         } catch (Exception e) {
             LOGGER.warn("Unexpected exception executing block. Trying to rollback...", e);
 
-            tryRollback();
+            tryRollback(handle);
 
             throw new DatabaseException("Unexpected exception when executing block \"" + block + "\"", e);
         } finally {
-            close();
+            close(handle);
         }
     }
 
-    private void tryRollback() {
+    private void tryRollback(Handle handle) {
```

Comment:
It is old code and I kept it as is.... But yes, it seems there is no need to have tryRollback()  method...